sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/BasicWriteStatsTracker.scala

```diff
@@ -106,6 +105,13 @@ class BasicWriteTaskStatsTracker(hadoopConf: Configuration)
 
   override def getFinalStats(): WriteTaskStats = {
     statCurrentFile()
+
+    // Reports bytesWritten and recordsWritten to the Spark output metrics.
+    Option(TaskContext.get()).map(_.taskMetrics().outputMetrics).foreach { outputMetrics =>
```

Comment:
Which test suite covers this newly added logic?
Previously, BasicWriteTaskStatsTrackerSuite seems to fail (8 of 10 test cases) because TaskContext is absent in the test suite.

Reply:
I don't know if there is any Spark job/stage UI test in the SQL module, and since this is a pretty trivial change, I just tested it manually. Previously, `BasicWriteTaskStatsTrackerSuite` failed with NPE because I didn't handle null TaskContext here. Now I wrap it with `Option` to make it null safe.